external-providers/generic-external-provider/pkg/server_configurations/nodejs/service_client.go

```diff
@@ -173,15 +174,86 @@ func (sc *NodeServiceClient) EvaluateReferenced(ctx context.Context, cap string,
 		return resp{}, fmt.Errorf("unable to get query info")
 	}
 
+	// Use FileSearcher to properly handle all filepath filtering sources:
+	// 1. Provider config constraints (dependency folders)
+	// 2. Rule scope constraints (from GetScopedFilepaths)
+	// 3. Condition filepaths (if they exist in the condition)
+	includedFilepaths, excludedFilepaths := cond.ProviderContext.GetScopedFilepaths()
+
+	// Determine base path - use first workspace folder if available
+	basePath := ""
+	if len(sc.BaseConfig.WorkspaceFolders) > 0 {
+		basePath = sc.BaseConfig.WorkspaceFolders[0]
+		// Remove file:// prefix if present
+		basePath = strings.TrimPrefix(basePath, "file://")
+	}
+
+	fileSearcher := provider.FileSearcher{
+		BasePath: basePath,
+		ProviderConfigConstraints: provider.IncludeExcludeConstraints{
+			IncludePathsOrPatterns: []string{},
```

Comment:
Do we not have the init config here?

Reply:
Missed that.  I've now added support for using included paths from the provider's init config, similar to how the java provider handles it.

   The nodejs provider now:
   1. Retrieves included paths from init config using \`provider.GetIncludedPathsFromConfig(c, false)\` during initialization
   2. Stores them in the service client struct
   3. Passes them to FileSearcher's ProviderConfigConstraints

   This ensures that provider-level path constraints (from \`ProviderSpecificConfig[IncludedPathsConfigKey]\`) are properly applied during file searching.